### PR TITLE
PMM-6826 Disk size in create API parameters.

### PR DESCRIPTION
**Pull request by BupycHuk**
https://github.com/percona/pmm-managed/pull/519
https://github.com/percona-platform/dbaas-controller/pull/91
https://github.com/percona/pmm/pull/613
https://github.com/percona-platform/dbaas-api/pull/62
https://github.com/Percona-Lab/pmm-submodules/pull/1220

### CODE REVIEW

File: service/k8sclient/k8client.go [hunk 1]
```diff
@@ -636,3 +573,29 @@ func (c *K8Client) setComputeResources(res *ComputeResources) *resourcesSpec {
...
 	return r
 }
+
+func (c *K8Client) getDiskSize(volumeSpec *common.VolumeSpec) int64 {
+	if volumeSpec == nil || volumeSpec.PersistentVolumeClaim == nil {
+		return 0
+	}
+	quantity, ok := volumeSpec.PersistentVolumeClaim.Resources.Requests[common.ResourceStorage]
+	if !ok {
+		return 0
```
Comment (askomorokhov) [inline service/k8sclient/k8client.go:583]: Looks contradictory that in resonance DiskSize can be zero and in the request cannot 
https://github.com/percona-platform/dbaas-api/pull/62/files#diff-3af72cef13d64190ae8c2b265df6707de29a26e75bfd5f09d98870e9effeb3eaR51

Should disc size be optional? 

And maybe the minimum should be at least 300MB. As when MySQL just start it require space for standard and temporary files:

```
48M /data/mysql/ib_logfile0
48M /data/mysql/ib_logfile1
192K /data/mysql/#ib_16384_0.dblwr
8.2M /data/mysql/#ib_16384_1.dblwr
41M /data/mysql/mysql.ibd
14M /data/mysql/undo_001
15M /data/mysql/undo_002
12M /data/mysql/ibtmp1
 ```

**Review by askomorokhov — APPROVED** [review]
Looks good to me with comments.

File: service/k8sclient/k8client.go (hunk 1 above)
Comment (BupycHuk) [reply service/k8sclient/k8client.go:583]: no, it should not be optional.
at the moment we don't validate any inputs, we will do it later. 
in UI minimum value is 1GB AFAIK.

**Review by percona-csalguero — APPROVED** [review]

File: service/cluster/xtra_db_cluster.go [hunk 2]
```diff
@@ -67,21 +67,23 @@ func (s *XtraDBClusterService) ListXtraDBClusters(ctx context.Context, req *cont
...
 		}
-		if cluster.PXC != nil {
-			params.Pxc = &controllerv1beta1.XtraDBClusterParams_PXC{
-				ComputeResources: &controllerv1beta1.ComputeResources{
-					CpuM:        cluster.PXC.ComputeResources.CPUM,
-					MemoryBytes: cluster.PXC.ComputeResources.MemoryBytes,
-				},
+		if cluster.PXC.ComputeResources != nil {
+			params.Pxc.ComputeResources = &controllerv1beta1.ComputeResources{
+				CpuM:        cluster.PXC.ComputeResources.CPUM,
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply service/cluster/xtra_db_cluster.go:79]: Here we do the same, but in another place https://github.com/percona-platform/dbaas-controller/blob/554d16f5e930d5d0d85bbda525f332cbc01eddde/service/k8sclient/k8client.go#L562

File: service/k8sclient/k8client.go (hunk 1 above)
Comment (AlekSi) [reply service/k8sclient/k8client.go:588]: FWIW, the minimal value of 1GB would be the sane minimal value in API too.

We have this problem with zero values in general (zero value in API – not set, not known, forgotten, or really set to zero?), so we can forbid zero value for disk size everywhere, that would be nice

File: service/k8sclient/k8client.go [hunk 3]
```diff
@@ -636,3 +573,29 @@ func (c *K8Client) setComputeResources(res *ComputeResources) *resourcesSpec {
...
+	}
+	return quantity.Value()
+}
+
+func (c *K8Client) volumeSpec(diskSize int64) *common.VolumeSpec {
+	if diskSize == 0 {
+		return nil
+	}
+	return &common.VolumeSpec{
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply service/k8sclient/k8client.go:597]: Please add `//nolint` comment or disable that linter if it produces too much noise, but please don't silently ignore it

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: service/k8sclient/k8client.go (hunk 1 above)
Comment (BupycHuk) [reply service/k8sclient/k8client.go:588]: it's in response in case we have some db clusters with not set volume values or configured another way

**Review by AlekSi — APPROVED** [review]
:shipit:

**Pull request merged**